tensorboard/plugins/core/core_plugin.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""TensorBoard core plugin package."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import functools
+import mimetypes
+
+import tensorflow as tf
+from werkzeug import utils
+from werkzeug import wrappers
+
+from tensorboard.backend import http_util
+from tensorboard.plugins import base_plugin
+
+
+class CorePlugin(base_plugin.TBPlugin):
+  """Core plugin for TensorBoard.
+
+  This plugin serves runs, configuration data, and static assets. This plugin
+  should always be present in a TensorBoard WSGI application.
+  """
+
+  plugin_name = 'core'
+  disable_path_prefix = True
+
+  def __init__(self, context):
+    """Instantiates CorePlugin.
+
+    Args:
+      context: A base_plugin.TBContext instance.
+    """
+    self._logdir = context.logdir
+    self._multiplexer = context.multiplexer
+    self._assets_zip_provider = context.assets_zip_provider
+
+  def is_active(self):
+    return True
+
+  def get_plugin_apps(self):
+    apps = {
+      '/audio': self._redirect_to_index,
+      '/data/logdir': self._serve_logdir,
+      '/data/runs': self._serve_runs,
+      '/events': self._redirect_to_index,
+      '/graphs': self._redirect_to_index,
+      '/histograms': self._redirect_to_index,
+      '/images': self._redirect_to_index,
+    }
+    if self._assets_zip_provider:
```

Comment:
What do you think about moving this loading logic into the plugin constructor, so that `get_plugin_apps` can be just a getter and not do disk I/O? (Perhaps this could obviate the `if plugin_name == 'core': raise e` in `application.py`, too?)